### PR TITLE
2.4.0

### PR DESCRIPTION
**Pull request by sjsrey**
this is prepping for a 2.4.0rc1 release

### CODE REVIEW

**Comment by sjsrey** [comment]
Looks like this will be stalled until we get a conda release of spopt.

**Comment by jGaboardi** [comment]
The [`staged-recipe`](https://github.com/conda-forge/staged-recipes/pull/13708) for `spopt` is now passing so once that PR is reviewed+merged we will have an `spopt-feedstock` up on `conda-forge`. At that point we should cut a `v0.1.0` release to include with the `v2.4.4` release of PySAL.

**Comment by sjsrey** [comment]
Once this is reviewed and merged, next ups is the release actions.

File: CHANGELOG.md
Comment (jGaboardi) [inline CHANGELOG.md:3]: We are leaving the CHANGELOG as `v2.3.0` for now and updating prior to the actual `v2.4.0` release, correct?

**Review by jGaboardi — APPROVED** [review]

File: CHANGELOG.md
Comment (sjsrey) [reply CHANGELOG.md:3]: Correct. This PR is just to get the unit tests dialed in with spopt. Next we move on to the release and change log update in another PR

**Comment by jGaboardi** [comment]
let's merge or wait for @knaaptime's review?

**Review by knaaptime — APPROVED** [review]

**Comment by sjsrey** [comment]
Can we remove codecov as it is more annoying than helpful. The imports are being tested, it just doesn't see that.

**Pull request merged**
